src/main/java/org/opensearch/plugin/insights/rules/model/SearchQueryRecord.java

```diff
@@ -39,10 +40,11 @@ public class SearchQueryRecord implements ToXContentObject, Writeable {
      */
     public SearchQueryRecord(final StreamInput in) throws IOException, ClassCastException {
         this.timestamp = in.readLong();
-        measurements = new HashMap<>();
-        in.readMap(MetricType::readFromStream, StreamInput::readGenericValue)
-            .forEach(((metricType, o) -> measurements.put(metricType, metricType.parseValue(o))));
+        measurements = new LinkedHashMap<>();
```

Comment:
Why it is needed to use `LinkedHashMap` ?

Reply:
Needed to make sure the writing and reading the object to IO works as expected. `testSerialize()` and some others were failing before this.